src/structures/ThreadChannel.js

```diff
@@ -0,0 +1,394 @@
+  }
+
+  _patch(data) {
+    super._patch(data);
+
+    /**
+     * The name of the thread
+     * @type {string}
+     */
+    this.name = data.name;
+
+    /**
+     * The ID of the parent channel to this thread
+     * @type {Snowflake}
+     */
+    this.parentID = data.parent_id;
+
+    /**
+     * Whether the thread is locked
+     * @type {boolean}
+     */
+    this.locked = data.thread_metadata.locked ?? false;
+
+    /**
+     * Whether the thread is active (false) or archived (true)
+     * @type {boolean}
+     */
+    this.archived = data.thread_metadata.archived;
+
+    /**
+     * The id of the member that created this thread
+     * @type {?Snowflake}
+     */
+    this.ownerID = data.owner_id;
+
+    /**
+     * How long in minutes after recent activity before the thread is automatically archived
+     * @type {number}
+     */
+    this.autoArchiveDuration = data.thread_metadata.auto_archive_duration;
+
+    /**
+     * The ID of the last message sent in this thread, if one was sent
+     * @type {?Snowflake}
+     */
+    this.lastMessageID = data.last_message_id;
+
+    /**
+     * The timestamp when the last pinned message was pinned, if there was one
+     * @type {?number}
+     */
+    this.lastPinTimestamp = data.last_pin_timestamp ? new Date(data.last_pin_timestamp).getTime() : null;
+
+    /**
+     * The ratelimit per user for this thread in seconds
+     * @type {number}
+     */
+    this.rateLimitPerUser = data.rate_limit_per_user ?? 0;
+
+    /**
+     * The timestamp the thread was last archived or unarchived at
+     * @type {?number}
+     */
+    this.archiveTimestamp = data.thread_metadata.archive_timestamp
+      ? new Date(data.thread_metadata.archive_timestamp).getTime()
+      : null;
+
+    /**
+     * The approximate count of messages in this thread
+     * <info>This value will not count above 50 even when there are more than 50 messages
+     * If you need an approximate value higher than this, use ThreadChannel#messages.cache.size</info>
+     * @type {number}
+     */
+    this.messageCount = data.message_count;
+
+    /**
+     * The approximate count of users in this thread
+     * <info>This value will not count above 50 even when there are more than 50 members</info>
+     * @type {number}
+     */
+    this.memberCount = data.member_count;
+
+    if (data.member && this.client.user?.id) this.members.add({ user_id: this.client.user.id, ...data.member });
+    if (data.messages) for (const message of data.messages) this.messages.add(message);
+  }
+
+  /**
+   * A collection of the guild member objects for each of this thread's members
+   * @type {Collection<Snowflake, GuildMember>}
+   * @readonly
+   */
+  get guildMembers() {
+    return this.members.cache.mapValues(member => member.guildMember);
+  }
+
+  /**
+   * The time the thread was last archived or unarchived at
+   * @type {Date}
+   * @readonly
+   */
+  get archivedAt() {
+    return new Date(this.archiveTimestamp);
```

Comment:
`archiveTimestamp` is nullable, so it should be checked here (and also update the type tag)